state/protocol/protocol_state/protocol_statemachine_test.go

```diff
@@ -13,63 +13,76 @@ import (
 	s.parentBlock = unittest.BlockHeaderFixture(unittest.HeaderWithView(s.parentProtocolState.CurrentEpochSetup.FirstView + 1))
 	s.candidate = unittest.BlockHeaderWithParentFixture(s.parentBlock)
+}
 
-	s.updater = newStateMachine(s.candidate.View, s.parentProtocolState)
+// ProtocolStateMachineSuite is a dedicated test suite for testing happy path state machine.
+type ProtocolStateMachineSuite struct {
+	BaseProtocolStateMachineSuite
+	stateMachine *protocolStateMachine
+}
+
+func (s *ProtocolStateMachineSuite) SetupTest() {
+	s.BaseProtocolStateMachineSuite.SetupTest()
+	var err error
+	s.stateMachine, err = newStateMachine(s.candidate.View, s.parentProtocolState)
+	require.NoError(s.T(), err)
 }
 
-// TestNewUpdater tests if the constructor correctly setups invariants for updater.
-func (s *UpdaterSuite) TestNewUpdater() {
-	require.NotSame(s.T(), s.updater.parentState, s.updater.state, "except to take deep copy of parent state")
-	require.Nil(s.T(), s.updater.parentState.NextEpoch)
-	require.Nil(s.T(), s.updater.state.NextEpoch)
-	require.Equal(s.T(), s.candidate.View, s.updater.View())
-	require.Equal(s.T(), s.parentProtocolState, s.updater.ParentState())
+// TestNewstateMachine tests if the constructor correctly setups invariants for protocolStateMachine.
+func (s *ProtocolStateMachineSuite) TestNewstateMachine() {
+	require.NotSame(s.T(), s.stateMachine.parentState, s.stateMachine.state, "except to take deep copy of parent state")
+	require.Nil(s.T(), s.stateMachine.parentState.NextEpoch)
+	require.Nil(s.T(), s.stateMachine.state.NextEpoch)
+	require.Equal(s.T(), s.candidate.View, s.stateMachine.View())
+	require.Equal(s.T(), s.parentProtocolState, s.stateMachine.ParentState())
 }
 
-// TestTransitionToNextEpoch tests a scenario where the updater processes first block from next epoch.
+// TestTransitionToNextEpoch tests a scenario where the protocolStateMachine processes first block from next epoch.
 // It has to discard the parent state and build a new state with data from next epoch.
-func (s *UpdaterSuite) TestTransitionToNextEpoch() {
+func (s *ProtocolStateMachineSuite) TestTransitionToNextEpoch() {
 	// update protocol state with next epoch information
 	unittest.WithNextEpochProtocolState()(s.parentProtocolState)
 
 	candidate := unittest.BlockHeaderFixture(
 		unittest.HeaderWithView(s.parentProtocolState.CurrentEpochSetup.FinalView + 1))
-	// since the candidate block is from next epoch, updater should transition to next epoch
-	s.updater = newStateMachine(candidate.View, s.parentProtocolState)
-	err := s.updater.TransitionToNextEpoch()
+	var err error
+	// since the candidate block is from next epoch, protocolStateMachine should transition to next epoch
+	s.stateMachine, err = newStateMachine(candidate.View, s.parentProtocolState)
 	require.NoError(s.T(), err)
-	updatedState, _, _ := s.updater.Build()
+	err = s.stateMachine.TransitionToNextEpoch()
+	require.NoError(s.T(), err)
+	updatedState, _, _ := s.stateMachine.Build()
```

Comment:
Lets make sure we test 
* all return values
* parent state has not been changed
```suggestion
	updatedState, stateID, hasChanges := s.stateMachine.Build()
	require.True(s.T(), hasChanges)
	require.NotEqual(s.T(), s.parentProtocolState.ID(), updatedState.ID())
	require.Equal(s.T(), updatedState.ID(), stateID)
	require.Equal(s.T(), s.parentProtocolState.ID(), s.stateMachine.ParentState().ID(), "should not modify parent protocol state")
```